engine/src/test/java/io/camunda/zeebe/engine/processing/incident/UserTaskIncidentTest.java

```diff
@@ -435,4 +435,323 @@ public void shouldResolveIncidentAfterCandidateUsersExpressionEvaluationFailed()
 
     assertThat(incidentResolved.getKey()).isEqualTo(incidentCreated.getKey());
   }
+
+  // --------------------------------------------------------------------------
+  // ----- TaskSchedule related tests
+
+  @Test
+  public void shouldCreateIncidentIfDueDateExpressionEvaluationFailed() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(processWithUserTask(u -> u.zeebeDueDateExpression("MISSING_VAR")))
+        .deploy();
+
+    // when
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    final var userTaskActivating =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATING)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementId(TASK_ELEMENT_ID)
+            .withElementType(BpmnElementType.USER_TASK)
+            .getFirst();
+
+    // then
+    assertIncidentCreated(processInstanceKey, userTaskActivating.getKey())
+        .hasErrorType(ErrorType.EXTRACT_VALUE_ERROR)
+        .hasErrorMessage(
+            "failed to evaluate expression 'MISSING_VAR': no variable found for name 'MISSING_VAR'");
+  }
+
+  @Test
+  public void shouldCreateIncidentIfDueDateExpressionOfInvalidType() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(processWithUserTask(u -> u.zeebeDueDateExpression("[1,2,3]")))
+        .deploy();
+
+    // when
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    final var userTaskActivating =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATING)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementId(TASK_ELEMENT_ID)
+            .withElementType(BpmnElementType.USER_TASK)
+            .getFirst();
+
+    // then
+    assertIncidentCreated(processInstanceKey, userTaskActivating.getKey())
+        .hasErrorType(ErrorType.EXTRACT_VALUE_ERROR)
+        .hasErrorMessage(
+            "Expected result of the expression '[1,2,3]' to be one of '[DATE_TIME, STRING]', but was 'ARRAY'");
+  }
+
+  @Test
+  public void shouldCreateIncidentIfDueDateExpressionIsEmptyString() {
```

Comment:
@korthout I've asked for a review from you as well since there is a slight difference in behavior with the new (`dueDate` and `followUpDate`) properties compared with the assignment-related properties.

I'm referring to what happens when an empty string is provided in an expression. In the assignment-related properties, an empty string is accepted as a valid expression output. For the `dueDate` and `followUpDate` properties, an empty string will result in an incident being raised. The reason is that, compared to an assignee or a candidate list, an empty string can't be a valid DateTime representation. 

So I would say that this difference in behavior is acceptable, but I'd like your opinion on it as well.

Reply:
Sorry for the late reply @koevskinikola. That is an interesting thought. So we would have the behaviors:
- valid due date string -> valid
- invalid due date string -> incident
- empty string -> incident (because not a valid date)
- null -> valid (meaning no date set)

🔧 Let's check this behavior with HTO. I imagine that an empty string can be interpreted as equivalent to `null`. But providing different ways to FAIL an expression evaluation strictly or to evaluate the expression allowing some leniency and gracefully allowing nullness is anyways a topic of debate. See
- https://github.com/camunda/feel-scala/issues/582

@saig0 Do you have any opinions on this?

Reply:
Sounds okay to me. :+1:

Reply:
@korthout just confirming that this isn't a problem from the HTO side.

So we can gracefully handle `null` / empty string evaluations for due/follow-up date expressions. :+1: